File: Top 50 Coding Interview Questions/05- Write code to Check if two strings are Anagram or not/twostringsareAnagramornot.cpp
#include<iostream>

using namespace std;

int main(){
    //Initialization variables
    char str1[100], str2[100];
    int first[26] = {
        0
    }, second[26] = {
        0
    }, c = 0,flag = 0;

    //Accepting inputs
    cout<< "Enter First String: ";
    gets(str1);
    cout<< "Enter second string: ";
    gets(str2);

    //Calculating frequencies of charaters in first string
    while(str1[c] != '\0')
    {
        first[str1[c] - 'a']++;
        c++;
    }

    c = 0;
    //calculating frequencies of characters in second string
    while(str2[c] != '\0')
    {
        second[str2[c] - 'a']++;
        c++;
    }
    //Checking if frequencies of both the strings are same or not
    for(c = 0; c < 26; c++)
    {
        if(first[c] != second[c])
          flag = 1;
    }

    //printing result
    if(flag == 0)
    {
        cout<< "Strings are anagram";
    }
    else{
        cout<< "Strings are not anagram";
    }
    return 0;
}

File: Top 50 Coding Interview Questions/05- Write code to Check if two strings are Anagram or not/twostringsareAnagramornot.py
#take user input
String1 = input('Enter the 1st string :')
String2 = input('Enter the 2nd string: ')
#check if length matches
if len(String1) != len(String2):
    #if false
    print('Strings are not anagram')
else:
    #sorted functions sort string by characters
    String1 = sorted(String1)
    String2 = sorted(String2)
#check if now strings matches
if String1 == String2:
#if true
    print('Strings are anagram')
else:
    print('Strings are not anagram') 